rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/deduplication/TbMsgDeduplicationNode.java

```diff
@@ -132,7 +155,7 @@ private void processDeduplication(TbContext ctx, EntityId deduplicationId) {
                         }
                     }
                     deduplicationResults.add(TbMsg.newMsg(
-                            config.getQueueName(),
+                            queueName,
```

Comment:
I think that for the DeduplicationStrategy first and last, we should use the queueName as well if it's not null. Otherwise, the queueName parameter from the rule node will not provide any effect for those strategies.